src/rdkafka_topic.c

```diff
@@ -681,7 +682,7 @@ static int rd_kafka_toppar_leader_update(rd_kafka_topic_t *rkt,
 
         rd_kafka_toppar_lock(rktp);
 
-        if (leader_epoch < rktp->rktp_leader_epoch) {
+        if (leader_epoch != -1 && leader_epoch < rktp->rktp_leader_epoch) {
```

Comment:
Please add a comment above for why we are excluding -1, as it may not indicate outdated metadata but rather than the broker is KIP-320-unsupported.